plugins/source/gcp/client/spec/spec.go

```diff
@@ -1,31 +1,62 @@
-package client
+package spec
 
 import (
-	"github.com/cloudquery/plugin-sdk/v4/scheduler"
-)
+	_ "embed"
 
-const (
-	defaultConcurrency = 50000
+	"github.com/cloudquery/plugin-sdk/v4/scheduler"
+	"google.golang.org/api/cloudresourcemanager/v1"
 )
 
 // Spec defines GCP source plugin Spec
 type Spec struct {
 	ProjectIDs                  []string           `json:"project_ids"`
-	ServiceAccountKeyJSON       string             `json:"service_account_key_json"`
 	FolderIDs                   []string           `json:"folder_ids"`
 	FolderRecursionDepth        *int               `json:"folder_recursion_depth"`
+	OrganizationIDs             []string           `json:"organization_ids"`
 	ProjectFilter               string             `json:"project_filter"`
+	OrganizationFilter          string             `json:"organization_filter"`
+	ServiceAccountKeyJSON       string             `json:"service_account_key_json"`
 	BackoffDelay                int                `json:"backoff_delay"`
 	BackoffRetries              int                `json:"backoff_retries"`
-	DiscoveryConcurrency        *int               `json:"discovery_concurrency"`
+	DiscoveryConcurrency        int                `json:"discovery_concurrency"`
 	EnabledServicesOnly         bool               `json:"enabled_services_only"`
-	OrganizationIDs             []string           `json:"organization_ids"`
-	OrganizationFilter          string             `json:"organization_filter"`
-	ServiceAccountImpersonation *CredentialsConfig `json:"service_account_impersonation"`
 	Concurrency                 int                `json:"concurrency"`
 	Scheduler                   scheduler.Strategy `json:"scheduler,omitempty"`
+	ServiceAccountImpersonation *CredentialsConfig `json:"service_account_impersonation"`
+}
+
+func (spec *Spec) SetDefaults() {
+	if spec.BackoffRetries <= 0 {
```

Comment:
If a user wanted `0` retries how would they do that?

Reply:
Addressed in d426712e9bbb647b13eadb92186f57f17f0e6039